x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/DataFrameAnalysisCustomFeatureIT.java

```diff
@@ -0,0 +1,270 @@
+    private static final String ALIAS_TO_KEYWORD_FIELD = "alias-to-keyword-field";
+    private static final String ALIAS_TO_NESTED_FIELD = "alias-to-nested-field";
+    private static final List<Boolean> BOOLEAN_FIELD_VALUES = List.of(false, true);
+    private static final List<Double> NUMERICAL_FIELD_VALUES = List.of(1.0, 2.0);
+    private static final List<Integer> DISCRETE_NUMERICAL_FIELD_VALUES = List.of(10, 20);
+    private static final List<String> KEYWORD_FIELD_VALUES = List.of("cat", "dog");
+
+    private String jobId;
+    private String sourceIndex;
+    private String destIndex;
+
+    @Before
+    public void setupLogging() {
+        client().admin().cluster()
+            .prepareUpdateSettings()
+            .setTransientSettings(Settings.builder()
+                .put("logger.org.elasticsearch.xpack.ml.dataframe", "DEBUG")
+                .put("logger.org.elasticsearch.xpack.core.ml.inference", "DEBUG"))
+            .get();
+    }
+
+    @After
+    public void cleanup() {
+        cleanUp();
+        client().admin().cluster()
+            .prepareUpdateSettings()
+            .setTransientSettings(Settings.builder()
+                .putNull("logger.org.elasticsearch.xpack.ml.dataframe")
+                .putNull("logger.org.elasticsearch.xpack.core.ml.inference"))
+            .get();
+    }
+
+    @Override
+    protected NamedXContentRegistry xContentRegistry() {
+        SearchModule searchModule = new SearchModule(Settings.EMPTY, Collections.emptyList());
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>(searchModule.getNamedXContents());
+        entries.addAll(new MlInferenceNamedXContentProvider().getNamedXContentParsers());
+        entries.addAll(new MlDataFrameAnalysisNamedXContentProvider().getNamedXContentParsers());
+        return new NamedXContentRegistry(entries);
+    }
+
+    public void testNGramCustomFeature() throws Exception {
+        initialize("test_ngram_feature_processor");
+        String predictedClassField = NUMERICAL_FIELD + "_prediction";
+        indexData(sourceIndex, 300, 50, NUMERICAL_FIELD);
+
+        DataFrameAnalyticsConfig config = new DataFrameAnalyticsConfig.Builder()
+            .setId(jobId)
+            .setSource(new DataFrameAnalyticsSource(new String[] { sourceIndex },
+                QueryProvider.fromParsedQuery(QueryBuilders.matchAllQuery()), null))
+            .setDest(new DataFrameAnalyticsDest(destIndex, null))
+            .setAnalysis(new Regression(NUMERICAL_FIELD,
+                BoostedTreeParams.builder().setNumTopFeatureImportanceValues(6).build(),
+                null,
+                null,
+                42L,
+                null,
+                null,
+                Collections.singletonList(new NGram(TEXT_FIELD, "f", new int[]{1, 2}, 0, 2, true))))
+            .setAnalyzedFields(new FetchSourceContext(true, new String[]{TEXT_FIELD, NUMERICAL_FIELD}, new String[]{}))
```

Comment:
I found this confusing at first because I thought the analyzed fields should include the ngram `f.x` fields and exclude the `TEXT_FIELD`. `setAnalyzedFields` is now poorly named it is more like `setFetchedFields`.

Is there a way of specifying which ngrams fields should be modelled or indeed for the output of any pre-processor which fields are used?

Reply:
`analyzed_fields` = All fields grabbed from docs. These fields are chosen for FULL analysis (including being processed)

There is no way of specifying feature inclusion for processed features. They are always included. This is for API simplicity.

Maybe renaming `analyzed_fields` to `fetched_fields` is proper. 

@dimitris-athanasiou ^ what do you think?